src/main/java/org/openrewrite/python/internal/PsiPaddingCursor.java

```diff
@@ -365,14 +368,14 @@ public Space consumeUntilExpectedNewline() {
     public Space consumeUntilExpectedWhitespace(String search) {
         WithStatus<Space> withStatus = consumeUntilFoundWithStatus(search);
         if (!withStatus.succeeded) {
-            throw new IllegalStateException("did not find pattern as expected");
+            throw new IllegalStateException("did not find pattern as expected;\n" + printDebuggingMessage("STOPPED HERE"));
```

Comment:
Q: Are we registering the exceptions as parse failures? An Exception should not be sent to the CLI for a parse issue.

Reply:
Our parsers should almost never allow an exception to escape from their parse methods.
Instead we catch any exceptions, then attach them as a marker to a `PlainText` representation of the file which failed to parse. This allows us to see what the exception was and the full contents of the file which failed to parse, and still be able to run plain-text recipes on it. 

@garyolsen here is [an example](https://github.com/openrewrite/rewrite/blob/7fe21c5c5964194c73745c5d42e61c68f2ced570/rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParser.java#L189) of another of our parsers doing this. Parse failures are smuggled out in the execution context, where whatever is running the parser may retrieve them later.